### PR TITLE
Allow a path (string) as a valid value for a bundle application resou…

### PR DESCRIPTION
**Pull request by ExternalReality**
…rce.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/bundlechanges/99/

File: handlers.go [hunk 1]
```diff
@@ -38,6 +38,25 @@ func handleApplications(add func(Change), services map[string]*charm.Application
 			charms[application.Charm] = change.Id()
 		}
 
+		resources := make(map[string]int)
+		localResources := make(map[string]string)
+		for resName, res := range application.Resources {
```
Comment (ExternalReality) [inline handlers.go]: Here we keep in mind that validation is done elsewhere, so we trust that either int or string will work.

**Comment by ExternalReality** [comment]
!!build!!

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/bundlechanges/100/

File: dependencies.tsv [hunk 2]
```diff
@@ -16,7 +16,7 @@ golang.org/x/crypto	git	aedad9a179ec1ea11b7064c57cbc6dc30d7724ec	2015-08-30T18:0
 golang.org/x/net	git	ea47fc708ee3e20177f3ca3716217c4ab75942cb	2015-08-29T23:03:18Z
 gopkg.in/check.v1	git	4f90aeace3a26ad7021961c297b22c42160c7b25	2016-01-05T16:49:36Z
 gopkg.in/errgo.v1	git	66cb46252b94c1f3d65646f54ee8043ab38d766c	2015-10-07T15:31:57Z
-gopkg.in/juju/charm.v6-unstable	git	503191ddf2590b15db9669c424660ed612d0a30a	2016-05-27T01:46:20Z
+gopkg.in/juju/charm.v6-unstable	git	687626a5da44d2f14f3f64865efce7aeda49b531	2016-05-27T01:46:20Z
```
Comment (frankban) [inline dependencies.tsv]: This does not seem ok. From CI: godeps: cannot update "/tmp/build/100/src/gopkg.in/juju/charm.v6-unstable": fatal: reference is not a tree: 687626a5da44d2f14f3f64865efce7aeda49b531

File: handlers.go [hunk 3]
```diff
@@ -38,6 +38,25 @@ func handleApplications(add func(Change), services map[string]*charm.Application
...
+			switch v := res.(type) {
+			case int:
+				resources[resName] = v
+			case string:
+				localResources[resName] = v
+			}
+		}
+
```
Comment (frankban) [inline handlers.go]: Is this newline required? [shrug]

File: handlers.go [hunk 4]
```diff
@@ -38,6 +38,25 @@ func handleApplications(add func(Change), services map[string]*charm.Application
...
+		if len(resources) == 0 {
+			resources = nil
+		}
+
```
Comment (frankban) [inline handlers.go]: Ditto.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/bundlechanges/101/

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/bundlechanges-merge

**Pull request merged**